xrmocap/core/evaluation/mvp_evaluation.py

```diff
@@ -1,67 +1,341 @@
+
+        # save predicted keypoints3d
+        precision = None
+        if is_main_process():
+            n_frame = len(kps3d_pred_list)
+            n_kps = kps3d_pred_list[0].shape[1]
+            kps3d_pred = np.full((n_frame, self.n_max_person, n_kps, 4),
+                                 np.nan)
+
+            for frame_idx, per_frame_kps3d in enumerate(kps3d_pred_list):
+                if len(per_frame_kps3d) > 0:
+                    n_valid_person, keypoints3d_pred_valid = \
+                        convert_result_to_kps([per_frame_kps3d])
+                    kps3d_pred[
+                        frame_idx, :n_valid_person] = keypoints3d_pred_valid
+
+            keypoints3d_pred = Keypoints(
+                dtype='numpy',
+                kps=kps3d_pred,
+                mask=kps3d_pred[..., -1] > 0,
+                convention=self.pred_kps3d_convention,
+                logger=self.logger)
+
+            # convert gt to keypoints3d
+            gt_n_person = np.max(
+                np.array([
+                    scene_keypoints.get_person_number()
+                    for scene_keypoints in self.dataset.gt3d
+                ]))
+            gt_n_kps = self.dataset.gt3d[0].get_keypoints_number()
+            kps3d_gt = np.full((self.gt_n_frame, gt_n_person, gt_n_kps, 4),
+                               np.nan)
+            mask_gt = np.full((self.gt_n_frame, gt_n_person, gt_n_kps), np.nan)
+
+            start_frame = 0
+            for keypoints3d_gt_scene in self.dataset.gt3d:
+
+                kps3d_gt_scene = keypoints3d_gt_scene.get_keypoints().numpy(
+                )  # [n_frame, n_person, n_kps, 4]
+                mask_gt_scene = keypoints3d_gt_scene.get_mask().numpy(
+                )  # [n_frame, n_person, n_kps]
+
+                n_frame, n_person, _, _ = kps3d_gt_scene.shape
+                end_frame = min(start_frame + n_frame, self.gt_n_frame)
+
+                kps3d_gt[start_frame:end_frame, :n_person, ...] = \
+                    kps3d_gt_scene[:end_frame-start_frame, ...]
+                mask_gt[start_frame:end_frame, :n_person, ...] = \
+                    mask_gt_scene[:end_frame-start_frame, ...]
+                start_frame = end_frame
+
+            keypoints3d_gt = Keypoints(
+                dtype='numpy',
+                kps=kps3d_gt,
+                mask=mask_gt,
+                convention=self.gt_kps3d_convention,
+                logger=self.logger)
+
+            if not is_train:
+                pred_file = os.path.join(self.output_dir, 'pred_kps3d.npz')
+                gt_file = os.path.join(self.output_dir, 'gt_kps3d.npz')
+
+                self.logger.info(f'Saving 3D keypoints to: {pred_file}')
+                keypoints3d_pred.dump(pred_file)
+                self.logger.info(f'Saving 3D keypoints to: {gt_file}')
+                keypoints3d_gt.dump(gt_file)
+
+        # quantitative evaluation and print result
```

Comment:
Add indent

Reply:
Fixed in [d58a15a](https://github.com/openxrlab/xrmocap/pull/47/commits/d58a15a9e9978d496b1e4abb214c07c375049e06).